src/js/client/signup.js

```diff
@@ -0,0 +1,145 @@
+// provide a front-end to /api/user/signup
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.express.couchuser.frontend.signup");
+
+    // Try to log in and display the results
+    gpii.express.couchuser.frontend.signup.submit = function (that, event) {
+        if (event) { event.preventDefault(); }
+        var name     = that.locate("name").val();
+        var email    = that.locate("email").val();
+        var password = that.locate("password").val();
+        var confirm  = that.locate("confirm").val();
+
+        // Our user handling library doesn't offer password confirmation, so we have to do it ourselves for now
+        if (password !== confirm) {
+            that.displayError(null, null, "The passwords you have entered don't match.");
+            return;
+        }
+
+        // This is not at all nice, but in order to make jQuery correctly send array data, we have to:
+        //
+        // a) set `traditional` to `true`, otherwise the roles variable comes through with brackets appended to it.
+        // b) send at least a two-item array (a one-item array becomes a string variable)
+        //
+        // Thanks jQuery!
+        var settings = {
+            type:        "POST",
+            url:         that.options.apiUrl + "/signup",
+            success:     that.displayReceipt,
+            error:       that.displayError,
+            traditional: true,
```

Comment:
I think we were puzzled by this before, and I noted that really for a POST, any data should be sent as the request body and hence it would be surprising if this become corrupted. But perhaps this is by virtue of this being a formenc submission rather than a JSON one - I imagine this is hardcoded within express-couchuser and not under our control?